packages/flutter/lib/src/gestures/recognizer.dart

```diff
@@ -202,7 +202,13 @@ abstract class PrimaryPointerGestureRecognizer extends OneSequenceGestureRecogni
         resolve(GestureDisposition.rejected);
         stopTrackingPointer(primaryPointer);
       } else {
-        handlePrimaryPointer(event);
+        try {
+          handlePrimaryPointer(event);
+        } catch (e) {
+          resolve(GestureDisposition.rejected);
+          stopTrackingPointer(primaryPointer);
+          rethrow;
```

Comment:
Another option would be surrounding each invocation of an event handler callback in the various GestureRecognizer subclasses with a wrapper that catches and logs exceptions.  The subclass can then continue with its state tracking code.

Do you think that would be cleaner?